### ISSUE TITLE
rancheros cloud-config embeded in ISO

### ISSUE THREAD

**Issue by segator** [rancher/os#2437]
**RancherOS Version: (ros os version)**
1.4.0

How can I build a rancherOS image with embeded cloud-config I want my developers can deploy rancherOS with custom cloudconfig with simple docker-machine for vmware or vbox

**Comment by niusmallnan** [comment]
You can try to add your custom config to `https://github.com/rancher/os/blob/master/os-config.tpl.yml`, then run `make` to build an ISO.

**Comment by segator** [comment]
I saw this file but is somehow possibly to include my config as a cloud-config
in the same way I load the system with pxe with the datasource parameter

**Comment by niusmallnan** [comment]
If you are using vsphere, you can try `vmwarevsphere-cloudinit` via docker-machine:
```
VCENTER_IP="192.168.1.113"
USERNAME="xxx"
PASSWORD="xxxx"
MEMORY="4096"
RANCHEROS_ISO="http://192.168.1.125/rancheros/rancheros-vmware.iso"
VCENTER_PORT="443"
DISK_SIZE="30000"
CLOUD_INIT="http://192.168.1.125/rancheros/ipxe/cloud-config.yaml"


MACHINE_NAME=$1

./rancher-machine create --driver vmwarevsphere \
    --vmwarevsphere-username=$USERNAME \
    --vmwarevsphere-password=$PASSWORD \
    --vmwarevsphere-memory-size=$MEMORY \
    --vmwarevsphere-boot2docker-url=$RANCHEROS_ISO \
    --vmwarevsphere-vcenter=$VCENTER_IP \
    --vmwarevsphere-vcenter-port=$VCENTER_PORT \
    --vmwarevsphere-disk-size=$DISK_SIZE \
    --vmwarevsphere-cloudinit=$CLOUD_INIT \
    $MACHINE_NAME
```

**Comment by segator** [comment]
yes for vsphere I have the problem solved, the problem that I have is regarding local environments with vmware workstation and vbox that seems docker machine doesn't have the option.
I don't know maybe with a vagrant creating a iso disk with the cloud-config.yml inside, will be loaded?

Thanks

**Comment by niusmallnan** [comment]
I am not a heavy user of vagrant. I dunno, but in general, it should have a cloud-init option.

**Comment by segator** [comment]
The question is more related to if rancheros will automatically read cloud config from attached iso or floppy

**Comment by niusmallnan** [comment]
I don't know which one you want to use.
All datasources in ros cloud-init are here: https://github.com/coreos/coreos-cloudinit/tree/master/datasource
